gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -53,6 +71,10 @@ public DatasetCleaner(FileSystem fs, Properties props) throws IOException {
     } catch (InvocationTargetException exception) {
       throw new IOException(exception);
     }
+    this.executor =
+        Executors
+            .newFixedThreadPool(Integer.parseInt(props.getProperty(DATASET_CLEAN_THREAD_POOL_SIZE,
```

Comment:
@chavdar using a cached thread pool poses the risk of having an unbounded number of threads running in the pool, which may happen if there are so many datasets to cleanup in a batch. This post talks about it and I think it makes sense http://dev.bizo.com/2014/06/cached-thread-pool-considered-harmlful.html.